### PR TITLE
Support "Provide a way to declare visibility of attributes defined by attr* methods in a single expression"

### PR DESCRIPTION
**Pull request by sanfrecce-osaka**
Since provided a way to declare visibility of attributes defined by attr* methods in a single expression after 3.0, I fixed the type signatures of below methods.

- `Module#private`
- `Module#private_class_method`
- `Module#protected`
- `Module#public`
- `Module#public_class_method`

cf. https://techlife.cookpad.com/entry/2020/12/25/155741#private-attr_reader-foo%E3%81%A8%E6%9B%B8%E3%81%91%E3%82%8B%E3%82%88%E3%81%86%E3%81%AB%E3%81%AA%E3%81%A3%E3%81%9F
cf. https://bugs.ruby-lang.org/issues/17314
cf. https://github.com/ruby/ruby/commit/eb8ea336d33af7e1dec4c17964c671c33cf75ce1
cf. https://github.com/ruby/ruby/commit/3a81daaf8dc037057d96b8e8cdc6ab1691e7e9d9

In addition, the type signatures for the following methods have been updated.

- `Module#protected`
- `Module#module_function`

cf. https://bugs.ruby-lang.org/issues/12495
cf. https://github.com/ruby/ruby/commit/75ecbda438670ec12641d1324d0e81a52ee02e0a

### CODE REVIEW

File: test/stdlib/Module_test.rb [hunk 1]
```diff
@@ -157,6 +270,48 @@ def bar; end
...
     )
+
+    assert_send_type(
+      "(Array[Symbol | String]) -> Array[Symbol | String]",
+      mod, :public, [:foo, "bar"]
+    )
+
+    assert_send_type(
+      "(Array[Symbol]) -> Array[Symbol]",
+      mod, :public, mod.attr_reader(:baz, :qux)
```
[the remark this answers is not in the document]
Comment (sanfrecce-osaka) [reply test/stdlib/Module_test.rb]: I fixed it with https://github.com/ruby/rbs/pull/1357/commits/64a65d986778cc6b55f96d1e5db96250f6f481e7

File: core/module.rbs [hunk 2]
```diff
@@ -1326,7 +1332,12 @@ class Module < Object
   #
   # To show a private method on RDoc, use `:doc:` instead of this.
   #
-  def protected: (*Symbol | String arg0) -> self
+  def protected: () -> nil
+    | (Symbol method_name) -> Symbol
+    | (Symbol, Symbol, *Symbol method_name) -> Array[Symbol]
+    | (string method_name) -> string
+    | (string | Symbol, string | Symbol, *string | Symbol method_name) -> Array[string | Symbol]
+    | (Array[string | Symbol]) -> Array[string | Symbol]
```
[the remark this answers is not in the document]
Comment (sanfrecce-osaka) [reply core/module.rbs]: I fixed it with [68c89e5](https://github.com/ruby/rbs/pull/1357/commits/68c89e545c72ded69cfe796bdd8b789805a6f1af)

File: test/stdlib/Module_test.rb [hunk 3]
```diff
@@ -74,6 +74,36 @@ def test_module_eval
...
+      mod, :module_function, "foo"
+    )
+    assert_send_type(
+      "(ToStr) -> ToStr",
+      mod, :module_function, ToStr.new("foo")
+    )
+
+    assert_send_type(
+      "(Symbol, String) -> Array[Symbol | String]",
+      mod, :protected, :foo, "bar"
```
[the remark this answers is not in the document]
Comment (sanfrecce-osaka) [reply test/stdlib/Module_test.rb]: I fixed it with [b1bc802](https://github.com/ruby/rbs/pull/1357/commits/b1bc8028bc3f75aee27efc27e0f93c9d91d3e881)

File: test/stdlib/Module_test.rb [hunk 4]
```diff
@@ -127,6 +157,89 @@ def bar; end
...
+
+    assert_send_type(
+      "(Array[Symbol | String]) -> Array[Symbol | String]",
+      mod, :private, [:foo, "bar"]
+    )
+
+
+    assert_send_type(
+      "(Array[Symbol]) -> Array[Symbol]",
+      mod, :private, mod.attr_reader(:baz, :qux)
```
[the remark this answers is not in the document]
Comment (sanfrecce-osaka) [reply test/stdlib/Module_test.rb]: I fixed it with [64a65d9](https://github.com/ruby/rbs/pull/1357/commits/64a65d986778cc6b55f96d1e5db96250f6f481e7)

File: test/stdlib/Module_test.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (sanfrecce-osaka) [reply test/stdlib/Module_test.rb]: I fixed it with [64a65d9](https://github.com/ruby/rbs/pull/1357/commits/64a65d986778cc6b55f96d1e5db96250f6f481e7)

**Review by pocke — APPROVED** [review]

**Pull request merged**